### ISSUE TITLE
[Android]安卓15如何使用呢以及响应重写格式问题

### ISSUE THREAD

**Issue by mydracula** [wanghongenpin/proxypin#362]
![e61f91f0b716b654b5d6cb9cd6232b5](https://github.com/user-attachments/assets/b7ebd6cf-83e3-48ee-92d6-0dbc55bc1dec)
![image](https://github.com/user-attachments/assets/54c60293-c1d3-4eed-88ba-34dbe1a01a0e)
![1c461d798d814416b3128f1a6a4e740](https://github.com/user-attachments/assets/6a162f2f-b647-4d02-9014-5be1db745a49)
这种没网的应该怎么解决
另外响应重写为什么没有格式化 我看别的响应没这个问题 我抓包美团h5订单详情就会出现这样
版本是1.1.5

https://github.com/user-attachments/assets/67d89554-9fba-423b-a0ed-bed74fa33b3e

**Comment by wanghongenpin** [comment]
ssl握手可能做了证书校验，你可以使用Firda绕过SSL Pinning检测,  请求重写填充是原报文，展示是经过json格式化了

**Comment by mydracula** [comment]
> 请求重写填充是原报文

那能不能响应帮我也格式化一下不然看着很费劲。我上个手机忘记是安卓几了安装上面几个模块和相同的方法就绕过了在15不知道怎么没效果

**Comment by wanghongenpin** [comment]
填充原报文是比较合理的，证书校验需要你自己研究，如果其他抓包软件，本软件抓不了 欢迎反馈

**Comment by mydracula** [comment]
> 填充原报文是比较合理的，证书校验需要你自己研究，如果其他抓包软件，本软件抓不了 欢迎反馈

那能不能格式化按钮呢 因为我用之前的版本就没有这个原报文的问题 我是昨天才更新的 就显示原报文 没有格式化 
![image](https://github.com/user-attachments/assets/78be5d81-48c2-473b-9f18-2473b6697afd)

**Comment by mydracula** [comment]
> 格式化后json会误导请求修改， 很多人反馈为什么请求修改不生效，因为不是实际报文
那就用实际报文修改呗 只是想要预览json

我又遇到了两个问题
问题一：抓包饿了么订单详情是乱码 url是https://alsc-buy2.ele.me/h5/mtop.alsc.eleme.order.detail.miniapp.build/1.0/5.0/ 这个我清楚记得以前还是可以的 不知道为什么突然不行了 reqable可以抓到 看截图。

[6009a43e5e7211793ad9e8ebc748e50.jpg](https://img.meituan.net/csc/db789c3a89b12205acf1f5c5a62d3cd7503458.jpg)
![49ac12085cfb40aab66a1b3dd4508a4.png](https://img.meituan.net/csc/cd4f2d35c4b7a8b3c0270673d18a7ef1681525.png)

问题二：我的url是https://xxxx.xxxx.xxxx/xxxx/giftList?type=1&sort=0&page=9&orderBy=desc&pageSize=20。我使用请求重写然后URL写的是正则,  https://xxxx.xxxx.xxxx/xxxx/giftList?(?:[^&]*&)*page=1(?:&[^&]*)*$ 为啥不行捏  这个也是旧版本可以 最新版本就不好使了 匹配不到

**Comment by wanghongenpin** [comment]
用1.1.6版本，我这可以， 重写匹配不上就把参数删了，只保留域名，多试试

![5e18d575f04ca967694cb74b5283e4ad_720](https://github.com/user-attachments/assets/84acd66f-45c8-4bed-be8d-947490e62f81)

**Comment by mydracula** [comment]
> 用1.1.6版本，我这可以， 重写匹配不上就把参数删了，只保留域名，多试试
> 
不能只保留域名啊，我只想匹配第一页。 假如是一个分页 他不是得判断第一页有没有数据假如有数据就会请求第二页你要是只匹配域名不是每一页都有数据一直刷新吗     现在我只想要第一页有数据

**Comment by wanghongenpin** [comment]
下载最新的包

![image](https://github.com/user-attachments/assets/24b6a804-042b-45a2-83c0-694adb6c782f)

**Comment by mydracula** [comment]
> 下载最新的包
> 
确实可以了，爱你哟。还有这个问题
假如我想在电脑修改这些请求，这个外部代理是不是类似小黄鸟的二级代理功能，然后 手机端是否展示抓包 这个开关是不是可能会有误导和歧义呢，因为关闭但是他还是会显示一些请求（虽然是报错的连接失败的），不知道这个开关是干嘛的
[img]https://img.meituan.net/csc/4957e87869d35ffcaaabb89d08f23291273418.jpg[/img]
我电脑本地开了个代理，我使用外部代理设置连我的本地（为什么不使用wifi代理，因为某些app会检测wifi代理），我就启用这个外部代理
这是启动的情况 正常抓到
[img]https://img.meituan.net/csc/50abc83dc66ad57b7457d976d58e08b41014633.png[/img]
这是关闭的情况 app就会网络错误 手机抓不到
[img]https://img.meituan.net/csc/e256294e69aa351bc32a1e60d536669d184069.jpg[/img]

**Comment by wanghongenpin** [comment]
是否展示抓包关闭，不会解包，原报文转发给电脑，错误请求展示也还好

**Comment by mydracula** [comment]
> 是否展示抓包关闭，不会解包，原报文转发给电脑，错误请求展示也还好

好像没有转发 抓不到了

**Comment by mydracula** [comment]
> 是否展示抓包关闭，不会解包，原报文转发给电脑，错误请求展示也还好

亲爱的作者大大，昨天系统更新后我又遇到证书问题就是那个ssl握手失败，然后上次是稀里糊涂的突然可以，这次我复现了一下大概知道是什么原因可以获得临时性抓包绕过ssl限制（重启就不行了），是一个很奇怪的问题，但是又有点无关本项目，我又不知道什么问题，不知道作者大大感不感兴趣

**Comment by farisyusry** [comment]
Yes. I faced this problem on samsung s25. But its working on my samsung s23. There look like some ssl url cant be intercept.

**Comment by wanghongenpin** [comment]
怎么临时绕过

> 亲爱的作者大大，昨天系统更新后我又遇到证书问题就是那个ssl握手失败，然后上次是稀里糊涂的突然可以，这次我复现了一下大概知道是什么原因可以获得临时性抓包绕过ssl限制（重启就不行了），是一个很奇怪的问题，但是又有点无关本项目，我又不知道什么问题，不知道作者大大感不感兴趣
